### ISSUE TITLE
Volumetric sampling bug/problems in Instant-NGP implemtation

### ISSUE THREAD

**Issue by dozeri83** [nerfstudio-project/nerfstudio#1099]
Hi @tancik , 
****
tl;dr - there is a problem in the volumetric sampler resulting in a memory crash.
The cause of this issue is the initial sigma_fn method in the ray marching.
*****

I have been experimenting for a while in different configurations and methods in nerfstudio and stumbled upon weird issue. 
First of all I am using instant-ngp  (with AABB contraction) implementation since I can compare it visually to the original instant-ngp.

I am using blender hotdog data set with aabb 1. (on nerfstudio - scene_scale 0.5)

nerfstudio seem to crash on GPU memory because of this line
https://github.com/nerfstudio-project/nerfstudio/blob/6baf31dc9a9d81e6b44478bd31e0c0bc575493b2/nerfstudio/pipelines/dynamic_batch.py#L74
num_samples_per_batch - was very small.

the reason num_samples_per_batch was very small is because on the first iteration of the train the volumetric sampler
rejected the samples because of  sigma_fn ( the samples had zero density so it rejected all of them)

https://github.com/nerfstudio-project/nerfstudio/blob/e605b3344ac623fd96e2458ef98f8fbc7a97d718/nerfstudio/model_components/ray_samplers.py#L458

I had 2 workarounds - 
1. replaced  sigma_fn with None.
2. replaced sigma_fn with None for the first 100 iteration and then replaced it with  self.get_sigma_fn(rays_o, rays_d).

workaround 1 work best for me (uniform sampler?) with better PSNR.

BTW - nerfacto model looked very bad on the blender hotdog data set. ( but this is a different issue)

**Comment by dozeri83** [comment]
Reconstruction:

ns-train instant-ngp --trainer.max-num-iterations 35000 --vis tensorboard  --experiment-name transforms_train_scaled_0_3333333 --output_dir output/hotdog --timestamp 23 --data nerf_sythetic/hotdog/  --orientation-method none --center-poses False --auto-scale-poses False --scene_scale 0.5

where I scaled the transforms file by 0.3333 (ngp scale) and used nerstudio loader:

[transforms_train_scaled_0_3333333.zip](https://github.com/nerfstudio-project/nerfstudio/files/10177371/transforms_train_scaled_0_3333333.zip)

**Comment by tancik** [comment]
I think the crux of the issue is that our models are set up to work with real-world data, not the blender data. (ie you can see a discussion in #806). The best solution is probably to have multiple models specific to different usecases, ie a "blender" version of instant-ngp and nerfacto (it would use a different sampling strategy, no pose opt, no appearance opt, no contraction...). On our end the question is how to prioritize improving real-world datasets for applications vs datasets like blender for research benchmarks.

**Comment by tancik** [comment]
I think the way to go about this is to create a new config, "instant-ngp-bounded" that contains these changes so that people can choose depending on the data they are working with.
